src/audio/codec_adapter/codec_adapter.c

```diff
@@ -401,49 +401,48 @@ int codec_adapter_copy(struct comp_dev *dev)
 		comp_update_buffer_consume(source, codec->cpd.consumed);
 	}
 
-	while (bytes_to_process) {
-		/* Proceed only if we have enough data to fill the lib buffer
-		 * completely. If you don't fill whole buffer
-		 * the lib won't process it.
-		 */
-		if (bytes_to_process < codec_buff_size) {
-			comp_dbg(dev, "codec_adapter_copy(): source has less data than codec buffer size - processing terminated.");
-			break;
-		}
-
-		/* Process only we have enough free data in the local
-		 * buffer. If we don't have enough free space process()
-		 * will override the data in the local buffer
-		 */
-		if (local_buff->stream.free < get_output_bytes(dev))
-			goto db_verify;
+	/* Proceed only if we have enough data to fill the lib buffer
```

Comment:
Would it make sense here to specify codec processing min/max watermarks in topology ? Otherwise looks good.

Reply:
@jgunn0262 not really. Because the watermark depends on dynamic parameters (e.g for MP3 it depends on mp3 version, which is a dynamic parameter). See `cadence_codec_get_samples` implementation in `src/audio/codec_adapter/codec/cadence.c`

Reply:
@dbaluta Isn't there a min/max fragment size for mp3 ? I think I get the gist of @jgunn0262 question.

Reply:
@lgirdwood is this question related to how we compute the buffer sizes? For now we make use of existent PCM parameters.